gam/spearman_distance.py

```diff
@@ -57,17 +60,21 @@ def spearman_squared_distance_legacy(r_1, r_2):
     return distance
 
 
-def pairwise_spearman_distance_matrix(rankings):
+def pairwise_spearman_distance_matrix(rankings, dask=False):
     """Returns Spearman Distances for the provided rankings
 
     Args:
         rankings (numpy.array): Normalized Attributions
+        dask (boolean): whether or not to use dask's implementation
 
     Returns:
         [array[array]]: Spearman Distance Matrix
     """
-    return pairwise_distances(rankings, metric=spearman_squared_distance)
-
+    if dask:
+        D = dask_pairwise_distances(da.array(rankings), rankings, metric=spearman_squared_distance)
```

Comment:
I would consider instead allowing the user to pass in dask arrays directly and this function can then call on the appropriate distance function in response. This way the user has control over how the data is read in and partitioned.